example/gmres/gmres.hpp

```diff
@@ -42,28 +42,35 @@
 //@HEADER
 */
 
-#include<math.h>
-#include"KokkosKernels_IOUtils.hpp"
-#include<Kokkos_Core.hpp>
-#include<Kokkos_Random.hpp>
-#include<KokkosBlas.hpp>
-#include<KokkosBlas3_trsm.hpp>
-#include<KokkosSparse_spmv.hpp>
-#include<KokkosSparse_Preconditioner.hpp>
+#include <math.h>
+#include "KokkosKernels_IOUtils.hpp"
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Random.hpp>
+#include <KokkosBlas.hpp>
+#include <KokkosBlas3_trsm.hpp>
+#include <KokkosSparse_spmv.hpp>
+#include <KokkosSparse_Preconditioner.hpp>
 
 ////////////////////////////////////////////////////////////////////////////////
 // libstdc++ half_t overloads
 ////////////////////////////////////////////////////////////////////////////////
 #if !KOKKOS_HALF_T_IS_FLOAT
-Kokkos::Experimental::half_t abs(Kokkos::Experimental::half_t arg) {
+Kokkos::Experimental::half_t fabs(Kokkos::Experimental::half_t arg) {
```

Comment:
This does not look like an overload of std::fabs since it is not defined in the same namespace.
I would suggest to implement the following instead:
```c++
namespace Kokkos {
namespace Experimental {
half_t = fabs(const half_t arg) {
...
}
}
}
```
The only annoying issue with this is that `Kokkos_Complex.hpp` does not define fabs but abs instead so we probably want to overload fabs for complex types: `Kokkos::complex<float>` and `Kokkos::complex<double>`.
In general these are things that we should implement in `Kokkos_ArtihTraits.hpp`

Reply:
Ok, I will wrap around the half_t abs function in arithtraits here.